### PR TITLE
Add operator reading with offsets

### PR DESCRIPTION
**Pull request by yurydelendik**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "wasmparser"
-version = "0.21.8"
+version = "0.21.9"
```
Comment (sunfishcode) [inline Cargo.toml:3]: This should be a minor version bump, as it adds new features.

File: src/readers/operators.rs [hunk 2]
```diff
@@ -49,6 +49,23 @@ impl<'a> OperatorsReader<'a> {
...
+    where
+        'a: 'b,
+    {
+        OperatorsIteratorWithOffsets {
+            reader: self,
+            err: false,
+        }
+    }
+
+    pub fn read_with_offset<'b>(&mut self) -> Result<(usize, Operator<'b>)>
```
Comment (sunfishcode) [inline src/readers/operators.rs]: This is just aesthetic preference, but I'd find this a little more natural if the `Operator<'b>` came first in the tuple, with the `usize` second.

File: src/readers/operators.rs [hunk 3]
```diff
@@ -104,3 +121,45 @@ impl<'a> Iterator for OperatorsIterator<'a> {
...
+    ///         ops
+    ///     );
+    /// }
+    /// ```
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.err || self.reader.eof() {
+            return None;
+        }
+        let result = self.reader.read_with_offset();
+        self.err = result.is_err();
```
Comment (sunfishcode) [inline src/readers/operators.rs:163]: Random idea: If `BinaryReader` had a way to say "skip to the end", then we could change this code to
```rust
    if result.is_err() {
        self.reader.skip_to_end();
    }
```
and then we wouldn't need these `self.err` fields in the various reader classes that have them, and less branching when calling `next`. Does that sound reasonable?

**Review by sunfishcode — APPROVED** [review]
Looks good!

File: Cargo.toml (hunk 1 above)
Comment (yurydelendik) [reply Cargo.toml:3]: AFAIK that's a minor bump for "0.x.y" versions

File: src/readers/operators.rs (hunk 3 above)
Comment (yurydelendik) [reply src/readers/operators.rs:163]: I tested this approach -- it resulted in 5% speed loss. I moved the commit into the [skip_to_end](https://github.com/yurydelendik/wasmparser.rs/tree/skip_to_end) branch.

**Pull request merged**